### DISCUSSION TITLE
There does appear to be a difference in versions

### DISCUSSION THREAD

**Discussion by Frogmore42** [arendst/Tasmota#10130] [Q&A] [answered]
I was working on what type of data I wanted to save from the information that my many Tasmota enabled devices generate, so I made a table of the Information page for all my devices.  I found 17 devices.

The oldest has a build date of 2017-10-07, so more than 3 years ago.  This is running version 5.8.0 with core/SDK 2_3_0/1.5.3(aec24ac9).  It has booted 1667 times and has 9133 flash writes.  That is about 8 flash writes a day and 1.5 boots per day.  This is a Sonoff TH with an SHT15 temp/humidity sensor attached to the vent fan in the bathroom.  I have a Node-RED automation that turns the fan on when the relative humidity rises above 70% and turns it off when it drops below 50% (or is constant for 30 min).  This has been working great for a very long time.  I also use HA-Bridge so Alexa can discover and control the fan (this is running code from before that capability got added to Tasmota).

The next really interesting device is also a Sonoff TH also with an SHT15.  It is in the basement and turns on a fan for 30s every 30m to circulate the air between utility rooms.  There is a heat pump water heater in one room and a dehumidifier in the other.  This one has 629,924 flash write, which was surprising.  It must be saving the state on every transition.  It only has 1657 reboots, so not rebooting significantly more frequently.  It is running 5.9.1 with the same core/SDK.

Now, there was a (several really) disruptions to the network several months ago, which made it impossible for the devices to contact the MQTT server.  This is probably responsible for some of the large boot count from some of the older devices, because that was the default behavior back then.

I also have two devices that are running the same Tasmota version (5.12.0i) but using different core/SDK versions (2.3.0 to 2.4.0).  The 2.4.0 one has half the number of reboots and is about the same age.

Tasmota 6.1.1.10 and/or code 2.4.2 brought a significant reduction in the number of boots, but this might have been when the behavior for unable to access MQTT changed.

Since I am using a mix of versions and many of them are very old, and I am changing a lot of things, I am going to update all of them with the current code as I integrate them into the new setup.

**Comment by ascillato** [comment, marked as answer]
SDK 2_3_0 and 2_4_x have a very nasty security bug. You should update ASAP.
SDK 2_5_x have a memory manager bug that keeps rebooting your devices.

That is why, all those old cores are not supported anymore. It is higly recommended to update to latest Tasmota following the update path explained in the docs and in the readme.

To be in the safe side, since your devices have a very old version, is to backup your settings before updating and also, after updating to reset your settings and configure your devices again.

Also, if you use Home Assistance, it is a good option to use the Native Integration.

**Reply by Frogmore42** [reply]
Yes, I have heard of the issues.  I was surprised that I had stuff that was as old as it was.  Part of the reason I haven't updated them, is that they are still working (even if they are rebooting fairly frequently).

I am not using Home Assistant.  I still remember when its use of retained messages caused lots of issues for many people.  I did set up OpenHAB, but it is an old version and I am not really that happy with it.  I was using it for its integration with influxdb, but the more I learn about influxdb, the less I like the integration.  In particular, OpenHAB does (or at least did) not have the concept of a missing signal from a sensor.  If a sensor goes offline, OpenHAB keeps the last value and keeps sending it.  It also required a significant amount of configuration to get s new Tasmota device added.  So much, that it was barely worth the effort.  My use case is probably somewhat special.  I have lots of sensors and a very limited amount of "hard" control (freeze protection, humidity control), so the benefits of HA or OpenHAB seem limited (and they require a large investment in time to understand how to "hold them right").

**Reply by Frogmore42** [reply]
But, thanks for the warning on the old cores.  I will update.
